### PR TITLE
fix(nuxt): disallow `write: false` for type templates

### PR DESCRIPTION
**Pull request by AndreyYolkin**
### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

As stated in https://github.com/nuxt/nuxt/blob/4e05650cde31ca73be4d14b1f0d23c7854008749/packages/kit/src/template.ts#L89-L92 `.d.ts` files always be written to the filesystem. This pull request introduces the NuxtTypeTemplate interface, where the write parameter for the addTypeTemplate function is omitted.

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/schema/src/types/nuxt.ts [hunk 1]
```diff
@@ -38,6 +38,8 @@ export interface ResolvedNuxtTemplate<Options = Record<string, any>> extends Nux
   dst: string
 }
 
+export type NuxtTypeTemplate<Options = Record<string, any>> = Omit<NuxtTemplate<Options>, 'write'>
```
Comment (danielroe) [inline packages/schema/src/types/nuxt.ts:41]: What do you think about instead having `write?: true` to preserve general compatibility?

Comment (AndreyYolkin) [reply packages/schema/src/types/nuxt.ts:41]: I was thinking about it too. In purpose of global compatibility we can, of course, keep the param, but set it to `true`

**Comment by AndreyYolkin** [comment]
Something unexpected happened with tests 😟

**Review by danielroe — APPROVED** [review]

**Comment by AndreyYolkin** [comment]
One day I'll write a PR title that won't have to change 🤞

**Pull request merged**